packages/react/src/components/Authenticator/shared/UserNameAlias.tsx

```diff
@@ -1,33 +1,67 @@
+import { useEffect } from 'react';
 import { I18n } from 'aws-amplify';
-import { getAliasInfoFromContext } from '@aws-amplify/ui';
+import {
+  ActorContextWithForms,
+  UserNameAlias,
+  countryDialCodes,
+  getActorContext,
+  getAliasInfoFromContext,
+} from '@aws-amplify/ui';
 
 import { useAuth } from '../../../hooks';
-import { TextField } from '../../../primitives';
+import { SelectField, TextField } from '../../../primitives';
 
 export interface UserNameAliasProps {
   handleInputChange?(event): void;
+  alias?: UserNameAlias;
   [key: string]: any;
 }
 
 export function UserNameAlias(props: UserNameAliasProps) {
-  const { handleInputChange, ...attrs } = props;
-  const [{ context }] = useAuth();
+  const { handleInputChange, alias, ...attrs } = props;
+  const [_state, send] = useAuth();
 
-  const { label, type, error } = getAliasInfoFromContext(context);
+  const actorContext: ActorContextWithForms = getActorContext(_state);
+  const { label, type, error } = getAliasInfoFromContext(_state.context, alias);
   const i18nLabel = I18n.get(label);
 
+  const isPhoneAlias = type === 'tel';
+
+  useEffect(() => {
+    isPhoneAlias &&
+      send({
+        type: 'CHANGE',
+        data: { name: 'country_code', value: actorContext.country_code },
+      });
+  }, []);
+
   return (
-    <TextField
-      type={type}
-      onChange={handleInputChange}
-      name="username"
-      required
-      placeholder={i18nLabel}
-      label={i18nLabel}
-      labelHidden={true}
-      errorMessage={error}
-      autoComplete="username"
-      {...attrs}
-    />
+    <>
+      {isPhoneAlias && (
+        <SelectField
+          name="country_code"
+          label="country code"
+          labelHidden={true}
+          defaultValue={actorContext.country_code}
+        >
+          {countryDialCodes.map((dialCode) => (
+            <option key={dialCode} value={dialCode}>
+              {dialCode}
+            </option>
+          ))}
+        </SelectField>
+      )}
+      <TextField
+        type={type}
+        name={alias ?? 'username'}
```

Comment:
Have you tested that SignIn/SignUp still works as expected with this change? Xstate looks specifically for the `username` value. I think changing the name here will cause errors.

Integ tests passing won't catch this since those don't actually perform the SignIn/signUp actions.

Reply:
Only the `signIn` service looks for `username` value and the `SignIn` component doesn't use `alias` because there will only ever be one input field. The `signUp` service looks for a name specific to the alias (email, username, phone_number) so that's why I use the `alias` as the name when it's specified by the parent component.

Reply:
Ok cool, thanks for clarifying that for me!

Reply:
It's a little tricky, it may be worth revisiting to make it clearer. Maybe doing something in the `SignIn` component where it's agnostic to the `name` prop, but always updates the state machine with that prop as `username`.